Add workflow to deploy plugin to WordPress.org

attackant opened the pull request:
- Fixes [1016](https://github.com/alleyinteractive/apple-news/issues/1016)

attackant committed:
Add workflow to deploy plugin to WordPress.org

This commit adds a new GitHub action workflow `deploy-to-wordpress.yml`. This workflow automates the plugin deployment to WordPress.org repository whenever a new tag is pushed to the develop branch. It includes all the necessary steps: setting up Node and PHP environments, installing dependencies, building the project, creating a GitHub release, creating and uploading plugin ZIP archive, and finally deploying the plugin to WordPress. This automation will significantly reduce human error and speed up the plugin release process.

renatonascalves commented on .github/workflows/deploy-to-wordpress.yml:
```diff
@@ -0,0 +1,91 @@
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
```
🤔 Do you want to deploy whenever a commit is merged to the main branch?

Looking at the rest of the file, you want to do that when creating a new tag/release.

renatonascalves replied:
Alley has Github actions to create automatic tag releases. You might want to use them and leave this only to deploy the plugin to WordPress.

renatonascalves commented on .github/workflows/deploy-to-wordpress.yml:
```diff
@@ -0,0 +1,91 @@
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
```
```suggestion
          php-version: 8.2
```

renatonascalves commented on .github/workflows/deploy-to-wordpress.yml:
```diff
@@ -0,0 +1,91 @@
+      - name: Get the version
+        id: packagejson
+        run: |
+          echo "::set-output name=version::$(jq -r '.version' package.json)"
```
🤔 Could you elaborate on why we are getting the version from the `package.json`? I'd assume this is not changed every time, no?

renatonascalves commented on .github/workflows/deploy-to-wordpress.yml:
```diff
@@ -0,0 +1,91 @@
+      - name: Create artifact
+        run: |
+          mkdir plugin-build
+          composer archive -vvv --format=zip --file="plugin-build/publish-to-apple-news"
```
I'd suggest reading this https://github.com/10up/action-wordpress-plugin-deploy?tab=readme-ov-file#excluding-files-from-deployment

This project doesn't have the files supported. That means many files would be deployed to WordPress, things you might now want to bundle in the deployed package.

Here is an example of how you can remove them: https://github.com/renatonascalves/wp-graphql-buddypress/blob/main/composer.json#L114-L132

Or you can add one of the deploy files supported by the action.

attackant committed:
Update .github/workflows/deploy-to-wordpress.yml

Co-authored-by: Renato Alves <19148962+renatonascalves@users.noreply.github.com>

attackant committed:
Update deployment workflow for version checking

The ".github/workflows/deploy-to-wordpress.yml" file has been modified to improve version control management. Changes include the addition of steps to get the current version, retrieve the latest tag, and check if the version has been incremented. These checks ensure that a new release has an incremented version number, preventing overwriting of previous releases. Removed the redundant "Get

attackant committed:
Update GitHub Action to abort on non-incremented versions

The GitHub Action workflow which deploys to WordPress is updated to exit with an error if the version is not incremented compared to the previous one. This change is introduced to prevent unintentional overwrites of existing plugin versions on WordPress. New 'Build' and 'Publish' steps will run only when version increment is detected, ensuring that each published version is unique.

attackant committed:
Update GitHub Actions workflow for better efficiency

This commit moves the GITHUB_TOKEN to the environment section of the release job and removes it from individual steps for better readability. It also improves the version comparison condition for more accuracy, and adds condition checks for each release-related task to only run if the version has incremented. This reduces unnecessary job executions, leading to quicker and more efficient workflow runs.

attackant committed:
add dry run

attackant committed:
add dry run

attackant committed:
Refactor deploy-to-wordpress workflow and update composer.json

This commit makes a major clean-up of the .github/workflows/deploy-to-wordpress.yml file. Now, it uses the 'alleyinteractive/.github/.github/workflows/built-release.yml' for building a draft release. It removes redundant steps that were repetitively checking for version increments and carrying out operations like setting up Node and PHP, build, etc., only if versions had been incremented. It allows composing the archive excluding specified files and directories. These changes simplify the configuration and make the process more efficient. Also, composer.json has been updated to exclude unnecessary files and directories when creating the archive.

attackant committed:
Add .distignore and update .deployignore

Added a new .distignore file and updated .deployignore to exclude files and directories we don't want included in the WordPress.org distribution or deployment. These changes aim to keep our deployment clean and free from unnecessary or unrelated files, such as configuration files for various developer tools or temporary/cache files created by the operating system or development environment.

attackant committed:
Move GITHUB_TOKEN environment variable to specific step

The GITHUB_TOKEN environment variable has been moved from the global environment of the workflow to the specific step where it is needed. This improves environment variable management and reduces unnecessary exposure of the GITHUB_TOKEN.

attackant committed:
Add built-release workflow and update deployment workflow

Added a new GitHub Actions workflow for building releases (`built-release.yml`). This workflow triggers on push events to `develop`, `main`, and `production` branches. It uses a predefined workflow from the `alleyinteractive` GitHub org, setting Node.js and PHP versions and enabling Composer installation.

Furthermore, the existing deployment to WordPress.org workflow (`deploy-to-wordpress.yml`) has been simplified. It no longer manually creates

attackant committed:
Update package.json version and script

The version of the 'publish-to-apple-news' package has been updated from "2.4.0" to "2.4.1". Additionally, two new scripts 'prerelease' and 'release' were added. The 'prerelease' script checks the node version, and the 'release' script creates a new release. These changes are made to facilitate the release process and ensure consistent node version during pre-release.

attackant committed:
Add release instructions to README

This commit adds a new section to the README file, detailing the steps required to release the plugin. It includes information on how the plugin utilises a built release workflow to compile and tag releases, as well as instructions for initiating a new release. This addition aims to provide clarity for future contributors or maintainers of the project.

attackant committed:
Remove Github actions workflows

Delete build.yml and release.yml which were previously used for managing automated builds and releases via Github Actions.

These workflows were removed because they are no longer in use. The automated building and releasing of the application is now handled by a separate CI/CD system. As such, these workflows are redundant and potentially confusing to future contributors.

srtfisher commented on .github/workflows/built-release.yml:
```diff
@@ -0,0 +1,18 @@
+    uses: alleyinteractive/.github/.github/workflows/built-release.yml@main
+    if: ${{ github.repository != 'alleyinteractive/create-wordpress-plugin' }}
+    with:
+      node: 16
+      php: '8.2'
```
This seems like it was copied over from create-wordpress-plugin. Just want to make sure these are the proper versions.

attackant replied:
Yes, those are correct, we have ticketed updating node for later.

srtfisher commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "publish-to-apple-news",
-  "version": "2.4.0",
+  "version": "2.4.1",
```
Bumping the plugin version here will make the plugin version out of step with the plugin's headers (which WordPress uses to manage the version). To mirror `create-wordpress-plugin`, we should also update the plugin's headers here:

- https://github.com/alleyinteractive/apple-news/blob/develop/apple-news.php#L17
- https://github.com/alleyinteractive/apple-news/blob/develop/readme.txt#L8

That is what the `npm run release` command should do, too.

attackant replied:
on it

attackant committed:
Update plugin version in apple-news.php

The version number of the 'Publish to Apple News' plugin has been updated from 2.4.0 to 2.4.1. This was done to reflect the latest changes or improvements made in the plugin.
